tests/integration/test_core.py

```diff
@@ -60,53 +53,48 @@ async def test_build_and_deploy(ops_test: pytest_operator.plugin.OpsTest, applic
 
 
 @pytest.mark.asyncio
-@pytest.mark.usefixtures("app_config")
-@pytest.mark.parametrize(
-    "app_config",
-    [
-        {
-            "db_host": "test_db_host",
-            "db_name": "test_db_name",
-            "db_user": "test_db_user",
-            "db_password": "test_db_password",
-        }
-    ],
-    indirect=True,
-    scope="function",
-)
-async def test_incorrect_db_config(ops_test: pytest_operator.plugin.OpsTest, application_name):
+@pytest.mark.abort_on_fail
+async def test_mysql_config(
+    request, ops_test: pytest_operator.plugin.OpsTest, application_name, kube_core_client
+):
     """
-    arrange: after WordPress charm has been deployed.
-    act: provide incorrect database info via config.
-    assert: charm should be blocked by WordPress installation errors, instead of lacking
-        of database connection info.
+    arrange: after WordPress charm has been deployed, and a mysql pod is deployed in kubernetes.
+    act: config the WordPress charm with the database configuration from a mysql pod.
+    assert: WordPress should be active.
     """
-    # Database configuration can retry for up to 60 seconds before giving up and showing an error.
-    # Default wait_for_idle 15 seconds in ``app_config`` fixture is too short for incorrect
-    # db config.
+    if not request.config.getoption("--test-db-from-config"):
+        pytest.skip()
     assert ops_test.model
-    await ops_test.model.wait_for_idle(idle_period=60)
-
-    for unit in ops_test.model.applications[application_name].units:
-        assert (
-            unit.workload_status == ops.model.BlockedStatus.name  # type: ignore
-        ), "unit status should be blocked"
-        msg = unit.workload_status_message
-        assert "MySQL error" in msg and (
-            "2003" in msg or "2005" in msg
-        ), "unit status message should show detailed installation failure"
+    application = ops_test.model.applications[application_name]
+    await application.set_config(
+        {
+            "db_host": kube_core_client.read_namespaced_pod(
+                name="mysql", namespace=ops_test.model_name
+            ).status.pod_ip,
+            "db_name": "wordpress",
+            "db_user": "wordpress",
+            "db_password": "wordpress-password",
```

Comment:
I'd use a constants for these parameters as they're also used when you create the mysql pod